base/abstractarray.jl

```diff
@@ -1436,15 +1436,15 @@ vcat(V::AbstractVector{T}...) where {T} = typed_vcat(T, V...)
 AbstractVecOrTuple{T} = Union{AbstractVector{<:T}, Tuple{Vararg{T}}}
 
 function _typed_vcat(::Type{T}, V::AbstractVecOrTuple{AbstractVector}) where T
-    n::Int = 0
+    n = 0
     for Vk in V
-        n += length(Vk)
+        n += length(Vk)::Union{Int,UInt}
```

Comment:
I'm not sure this is safe to assume.

Reply:
I'll change it to `Integer`. That will leave some `promote_rule` invalidations but it maybe be better than nothing.

Reply:
Or do you think we can safely convert-and-assert? Are you wondering about supporting arrays with more elements than `typemax(Int)`?